sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/api/FieldMatcher.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * SonarSource Kotlin
+ * Copyright (C) 2018-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.api
+
+import org.jetbrains.kotlin.descriptors.ClassifierDescriptor
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
+import org.jetbrains.kotlin.resolve.descriptorUtil.getAllSuperClassifiers
+
+fun interface FieldMatcher {
+    fun matches(node: KtNameReferenceExpression, bindingContext: BindingContext): Boolean
+}
+
+class FieldMatcherBuilderContext {
+    /**
+     * Field names to match against, or empty to match against all field names.
+     */
+    var names: Set<String> = emptySet()
+
+    /**
+     * Declaring types to match against, or empty to match against all declaring types.
+     */
+    var definingSupertypes: Set<String> = emptySet()
+
+    internal val fieldMatchers = mutableSetOf<FieldMatcher>()
+
+    fun fieldMatcher(initializer: FieldMatcherBuilderContext.() -> Unit) {
+        fieldMatchers += createFieldMatcher(initializer)
+    }
+}
+
+fun fieldMatcher(initializer: FieldMatcherBuilderContext.() -> Unit) =
```

Comment:
If this is a way to create a `FieldMatcher`, that's better to start with a Capital letter. So you `pretend` to be a constructor

Reply:
Fixed.